### PR TITLE
Fix https://github.com/nestjs/nest-cli/issues/1995

### PR DESCRIPTION
**Pull request by hkjeffchan**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/nestjs/nest/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
Bug fix

```
[x] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Other... Please describe:
```

## What is the current behavior?

webpack cache is not created using `nest build` if cache is set in webpack.config.js 

Issue Number: 1995

## What is the new behavior?
webpack cache is created using `nest build` if cache is set in webpack.config.js 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: lib/compiler/webpack-compiler.ts [hunk 1]
```diff
@@ -140,6 +140,7 @@ export class WebpackCompiler {
         return process.exit(1);
       }
       console.log(statsOutput);
+      compiler.close(() => {});
```
Comment (kamilmysliwiec) [inline lib/compiler/webpack-compiler.ts:143]: Are you sure we want to close the compiler - even if it works in the watch mode?

Comment (hkjeffchan) [reply lib/compiler/webpack-compiler.ts]: you are correct. should not close in watch mode... i'll update the PR

**Comment by hkjeffchan** [comment]
Updated the PR.

File: lib/compiler/webpack-compiler.ts [hunk 2]
```diff
@@ -140,6 +140,9 @@ export class WebpackCompiler {
         return process.exit(1);
       }
       console.log(statsOutput);
+      if (!watchMode && !watch) {
+        compiler.close(() => {});
```
Comment (kamilmysliwiec) [inline lib/compiler/webpack-compiler.ts:144]: Why can't we move this up - above the process.exit()?

Comment (hkjeffchan) [reply lib/compiler/webpack-compiler.ts:144]: or just leave it here at line 143

Comment (hkjeffchan) [reply lib/compiler/webpack-compiler.ts:144]:
```
      if (!err && !stats!.hasErrors()) {
        if (!onSuccess) {
          assetsManager.closeWatchers();
        } else {
          onSuccess();
        }
      } else if (!watchMode && !watch) {
        console.log(statsOutput);
        return process.exit(1);
      }
      console.log(statsOutput);
      if (!watchMode && !watch) {
        compiler.close(() => {});
      }
```

or 

```
      if (!err && !stats!.hasErrors()) {
        if (!onSuccess) {
          assetsManager.closeWatchers();
        } else {
          onSuccess();
        }
        if (!watchMode && !watch) {
          compiler.close(() => {});
        }
      } else if (!watchMode && !watch) {
        console.log(statsOutput);
        return process.exit(1);
      }
```

Comment (hkjeffchan) [reply lib/compiler/webpack-compiler.ts:144]: @kamilmysliwiec Would you check if you can merge this now?

**Pull request closed without merging**